### PR TITLE
Component structure implement routing

### PR DESCRIPTION
**Pull request by hedita**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,7 +1,21 @@
...
+import ConfigurationPage from "../pages/ConfigurationPage/ConfigurationPage";
 
 const App = () => {
-
+  <BrowserRouter>
+    <Routes>
+      <Route path="/Add" element={<AddPage />} />
+      <Route path="/All" element={<AllPage />} />
+      <Route path="/Completed" element={<CompletedPage />} />
+      <Route path="/UnCompleted" element={<UnCompletedPage />} />
```
Comment (blacksrc) [inline src/App.jsx]: These routes should use the same component as they have the same design.

File: src/App.jsx [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+      <Route path="/ConfigurationPage" element={<ConfigurationPage />} />
```
Comment (blacksrc) [inline src/App.jsx]: Use `hyphen case` (`kebab case`) for urls.

File: components/Add/Add.jsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const Add = () => {
+  return
```
Comment (blacksrc) [inline components/Add/Add.jsx]: return something so you can verify that routing works.

File: components/All/All.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const All = () => {
+  return
```
Comment (blacksrc) [inline components/All/All.jsx]: return something so you can verify that routing works.

File: components/Completed/Completed.jsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+
+const Completed = () => {
+  return (
+    "hi"
+  );
```
Comment (blacksrc) [inline components/Completed/Completed.jsx]: return something meaningful so you can verify that routing works.

File: components/Configuration/Configuration.jsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const Configuration = () => {
+  return
```
Comment (blacksrc) [inline components/Configuration/Configuration.jsx]: return something so you can verify that routing works.

File: components/UnCompleted/UnCompleted.jsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const UnCompleted = () => {
+  return
```
Comment (blacksrc) [inline components/UnCompleted/UnCompleted.jsx]: return something so you can verify that routing works.

File: src/App.jsx [hunk 8]
```diff
@@ -1,7 +1,21 @@
...
+import AddPage from "../pages/AddPage/AddPage";
+import AllPage from "../pages/AllPage/AllPage";
+import CompletedPage from "../pages/CompletedPage/CompletedPage";
+import UnCompletedPage from "../pages/UnCompletedPage/UnCompletedPage";
+import ConfigurationPage from "../pages/ConfigurationPage/ConfigurationPage";
 
 const App = () => {
-
+  <BrowserRouter>
+    <Routes>
```
Comment (blacksrc) [inline src/App.jsx]: If someone opens your website. They directly open `/`. It means that nothing will be loaded for them, Define a `/` route and load either `Add` or `All` component.

**Comment by blacksrc** [comment]
Also implement the general design and styling. The box in the middle with the `Todo` title and the tab links.
image

File: src/App.jsx (hunk 8 above)
Comment (hedita) [reply src/App.jsx]: Done.

File: components/UnCompleted/UnCompleted.jsx (hunk 7 above)
Comment (hedita) [reply components/UnCompleted/UnCompleted.jsx]: Done.

File: components/Configuration/Configuration.jsx (hunk 6 above)
Comment (hedita) [reply components/Configuration/Configuration.jsx]: Done.

File: components/Completed/Completed.jsx (hunk 5 above)
Comment (hedita) [reply components/Completed/Completed.jsx]: Done.

File: components/All/All.jsx (hunk 4 above)
Comment (hedita) [reply components/All/All.jsx]: Done.

File: components/Add/Add.jsx (hunk 3 above)
Comment (hedita) [reply components/Add/Add.jsx]: Done.

File: src/App.jsx (hunk 2 above)
Comment (hedita) [reply src/App.jsx]: Done.

File: src/App.jsx (hunk 1 above)
Comment (hedita) [reply src/App.jsx]: Done.

File: src/components/Navbar/Navbar.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import "./Navbar.scss";
+import { Link } from "react-router-dom";
+
+const Navbar = () => {
+  return (
+    <nav className="navbar">
+      <ul className="filter-list">
+        <li className="filter-item">
+          <Link className="navbar-item" to="/Add">
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]: `to` value should be lower case as it's defined in the router.

File: src/components/Navbar/Navbar.jsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+            Add
+          </Link>
+        </li>
+        <li className="filter-item">
+          <Link className="navbar-item" to="/">
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx:15]: `to` value should be lower case as it's defined in the router.

File: src/components/Navbar/Navbar.jsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+            All
+          </Link>
+        </li>
+        <li className="filter-item">
+          <Link className="navbar-item" to="/Completed">
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]: `to` value should be lower case as it's defined in the router.

File: src/components/Navbar/Navbar.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            Completed
+          </Link>
+        </li>
+        <li className="filter-item">
+          <Link className="navbar-item" to="/Uncompleted">
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]: `to` value should be lower case as it's defined in the router.

File: src/components/Navbar/Navbar.jsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+            Uncompleted
+          </Link>
+        </li>
+        <li className="filter-item">
+          <Link className="navbar-item" to="/Configuration">
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.jsx]: `to` value should be lower case as it's defined in the router.

File: src/components/Navbar/Navbar.scss [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+  
+.content {
+  background-color: $sixth-color;
+  display: grid;
+  place-items: center;
+  border-radius: $sizing-small;
+}
+
+.navbar {
+	padding-right: 95px;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: Indentation.

File: src/components/Navbar/Navbar.scss [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.filter-list {
+  display: grid;
+	grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: Indentation.

File: src/components/Navbar/Navbar.scss [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+.filter-list {
+  display: grid;
+	grid-template-columns: 1fr 1fr 1fr 1fr 1fr; 
+  margin-right: $spacing-small;
+}
+ 
+.filter-item {
+  color: $primary-color;
+	margin-left: $spacing-xxlarge;
+	text-align: center;
```
Comment (blacksrc) [inline src/components/Navbar/Navbar.scss]: Indentation.

File: src/App.jsx [hunk 17]
```diff
@@ -1,7 +1,36 @@
...
+        <div className="site-wrapper">
+          <div className="container">
+            <main className="content">
+              <Header />
+              <Navbar />
+              <Routes>
+                <Route path="/add" element={<AddPage />} />
+                <Route path="/" element={<AllPage />} />
+                <Route path="/completed" element={<CompletedPage />} />
+                <Route path="/uncompleted" element={<UnCompletedPage />} />
```
Comment (blacksrc) [inline src/App.jsx]: These 3 routes should use the same component. The reason is that, they responsibility is to render the list of todos. The only thing that is different is the data inside the component should be filtered.

Refer to the voice I put in the group.

File: src/components/Navbar/Navbar.jsx (hunk 9 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 10 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx:15]: It is a / and it does not have lower or upper case. Am I true?

File: src/components/Navbar/Navbar.jsx (hunk 11 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 12 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 13 above)
Comment (hedita) [reply src/components/Navbar/Navbar.jsx]: Done.

File: src/App.jsx (hunk 17 above)
Comment (hedita) [reply src/App.jsx]: Done.

File: src/components/Navbar/Navbar.jsx (hunk 10 above)
Comment (blacksrc) [reply src/components/Navbar/Navbar.jsx:15]: Correct. My bad 😅

File: src/pages/AllCompletedUncompleted/AllCompletedUncompleted.jsx [hunk 18]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const AllCompletedUncompleted = () => {
```
Comment (blacksrc) [inline src/pages/AllCompletedUncompleted/AllCompletedUncompleted.jsx]: Call this component something like `TodoList`.

File: src/App.jsx [hunk 19]
```diff
@@ -1,7 +1,31 @@
...
+    <BrowserRouter>
+      <>
+        <div className="site-wrapper">
+          <div className="container">
+            <main className="content">
+              <Header />
+              <Navbar />
+              <Routes>
+                <Route path="/add" element={<AddPage />} />
+                <Route path="/" element={<AllCompletedUncompleted />} />
```
Comment (blacksrc) [inline src/App.jsx]: You should not delete all the routes, Now `uncompleted` and `completed` links are broken. Bring them back and render the same component. (`TodoList`).

The idea of having just one component comes from the fact that they all have the same ui and rendering logic. So, all of them should render the same component but should have different routes. Later on, we get to that step that we pass extra parameter to url and get that inside component and apply the filtering to the list.

**Comment by blacksrc** [comment]
To fix indentation, configure prettier properly.
Add the following properties to the `.prettierrc` file.
```json
{
  "tabWidth": 2,
  "useTabs": false,
  "singleQuote": false
}
```
Also change `format` script from: this
```bash
prettier --write \"src/**/*.{js,jsx}\"
```
To this:
```bash
prettier --write \"src/**/*.{js,jsx,scss}\"
```
Then, run:
```bash
npm run format
```
This should format it properly.

> Explanation: 
> `tabWidth`: https://prettier.io/docs/en/options.html#tab-width
> `useTabs`: https://prettier.io/docs/en/options.html#tabs
> `singleQuote`: https://prettier.io/docs/en/options.html#quotes
> Adding `scss` to the files extension tells prettier to format that file type as well.

For more config options you can read this page: https://prettier.io/docs/en/options.html

File: src/App.jsx (hunk 19 above)
Comment (hedita) [reply src/App.jsx]: Done.

File: src/pages/AllCompletedUncompleted/AllCompletedUncompleted.jsx (hunk 18 above)
Comment (hedita) [reply src/pages/AllCompletedUncompleted/AllCompletedUncompleted.jsx]: Done.

File: src/components/Navbar/Navbar.scss (hunk 16 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 15 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

File: src/components/Navbar/Navbar.scss (hunk 14 above)
Comment (hedita) [reply src/components/Navbar/Navbar.scss]: Done.

**Pull request merged**